### ISSUE TITLE
`.latest()` ORM method is broken: Cannot filter a query once a slice has been taken.

### ISSUE THREAD

**Issue by MDziwny** [deschler/django-modeltranslation#591]
The last release (0.17) of `django-modeltranslation` has broken the usage of [`.latest()`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#latest) with django 3.2.

``` python
from django.db import models
from modeltranslation.translator import TranslationOptions, register

class TranslatedModel(models.Model):
    translated_field = models.CharField(max_length=255, null=True)

@register(models.TranslatedModel)
class TranslatedModelTranslationOptions(TranslationOptions):
    fields = ("translated_field",)
```

Then in a django shell:
``` bash
>>> from bug_app.models import TranslatedModel
>>> TranslatedModel.objects.latest("id")
...
AssertionError: Cannot filter a query once a slice has been taken.
```

**Comment by last-partizan** [comment]
What version are you using?

**Comment by MDziwny** [comment]
The exact versions are:
* Python 3.8.6
* Django 3.2.0
* django-modeltranslation 0.17

**Comment by last-partizan** [comment]
It is working for me.

I added a test in `591-test` branch, it's working fine. Can you try to reproduce it?

```
> AssertionError: Cannot filter a query once a slice has been taken.
```

Show your query for objects, maybe your manager is slicing something?

**Comment by MDziwny** [comment]
The value returned by `.latest()` needs to be assigned to a variable to trigger the error in the test:

```
diff --git a/modeltranslation/tests/tests.py b/modeltranslation/tests/tests.py
index 4b7885d..f2e9e3c 100644
--- a/modeltranslation/tests/tests.py
+++ b/modeltranslation/tests/tests.py
@@ -2654,7 +2654,7 @@ class TestManager(ModeltranslationTestBase):
         manager = models.ManagerTestModel.objects
         manager.create(title='more_de', visits_en=1, visits_de=2)
         manager.create(title='more_en', visits_en=2, visits_de=1)
-        manager.latest("id")
+        model_instance = manager.latest("id")
 
     def assert_fallback(self, method, expected1, *args, **kwargs):
         transform = kwargs.pop('transform', lambda x: x)
```

**Comment by last-partizan** [comment]
It changes nothing, still not failing.

**Comment by MDziwny** [comment]
Sorry, I was a bit surprised that just assigning the variable triggers the error, actually, the good diff is:

```diff --git a/tox.ini b/tox.ini
index 4325a3e..d1c9e5d 100644
--- a/tox.ini
+++ b/tox.ini
@@ -8,6 +8,7 @@ distribute = False
 envlist =
     py{36,37,38}-2.2,
     py{36,37,38}-3.0,
+    py{36,37,38}-3.2,
 
 [testenv]
 downloadcache = {toxworkdir}/_download/
@@ -21,5 +22,6 @@ basepython =
 deps =
     2.2: Django==2.2.*
     3.0: Django>=3.0a1,<3.1
+    3.2: Django>=3.2
     Pillow
     six
```

**Comment by last-partizan** [comment]
Yes, combination with django 3.2 it indeed fails.

**Comment by last-partizan** [comment]
@314eter can you investigate this please?

**Comment by MDziwny** [comment]
I have opened a MR with a fix proposal, but I'm not sure whether the usage of an iterator with map was on purpose.

**Comment by last-partizan** [comment]
I think it wasn't. In versions befor 3.2 there was `*args` syntax, and in 3.2 it was changed to `args`, and so map there was obvious error. (Not so obvious as it turned out :))

**Comment by last-partizan** [comment]
Released version 0.17.1, thank you for the fix
